commit 780183d3c3b8ddc6172e215039b8cb97b2ca6d99
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:24 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-curl-fallback.diff

diff --git a/src/port1.0/portfetch.tcl b/src/port1.0/portfetch.tcl
index 9503032ec..195e5fb91 100644
--- a/src/port1.0/portfetch.tcl
+++ b/src/port1.0/portfetch.tcl
@@ -556,7 +556,19 @@ proc portfetch::fetchfiles {args} {
                     set fetched 1
                     break
                 } on error {eMessage} {
-                    ui_debug [msgcat::mc "Fetching distfile failed: %s" $eMessage]
+                    ui_warn [msgcat::mc "Fetching distfile failed: %s" $eMessage]
+                    ui_debug "complete command:"
+                    ui_debug "curl fetch {*}$fetch_options $file_url \"${distpath}/${distfile}.TMP\""
+                    if {${eMessage} eq "gnutls_handshake() failed: Handshake failed"} {
+                        if {![catch {system "curl -L $file_url -o \"${distpath}/${distfile}.TMP\""} err]} {
+                            ui_debug [msgcat::mc "Fetching distfile succeeded via curl"]
+                            file rename -force "${distpath}/${distfile}.TMP" "${distpath}/${distfile}"
+                            set fetched 1
+                            break
+                        } else {
+                            ui_warn [msgcat::mc "Fetching distfile failed also via curl: %s" $err]
+                        }
+                    }
                     set lastError $eMessage
                 } finally {
                     file delete -force "${distpath}/${distfile}.TMP"
